oap-server/server-storage-plugin/storage-elasticsearch-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/elasticsearch/base/MetricsEsDAO.java

```diff
@@ -52,50 +52,49 @@ protected MetricsEsDAO(ElasticSearchClient client,
 
     @Override
     public List<Metrics> multiGet(Model model, List<Metrics> metrics) {
-        Map<String, List<Metrics>> groupIndices
-            = metrics.stream()
-                     .collect(
-                         groupingBy(metric -> {
+        Map<String, List<Metrics>> groupAliasIndices = new HashMap<>();
+        Map<String, List<Metrics>> groupIndices = new HashMap<>();
+        metrics.forEach(metric -> {
                              if (model.isTimeRelativeID()) {
                                  // Try to use with timestamp index name(write index),
-                                 // if latest cache shows this name doesn't exist,
-                                 // then fail back to template alias name.
-                                 // This should only happen in very rare case, such as this is the time to create new index
-                                 // as a new day comes, and the index cache is  pseudo real time.
-                                 // This case doesn't affect the result, just has lower performance due to using the alias name.
-                                 // Another case is that a removed index showing existing also due to latency,
-                                 // which could cause multiGet fails
-                                 // but this should not happen in the real runtime, TTL timer only removed the oldest indices,
-                                 // which should not have an update/insert.
                                  String indexName = TimeSeriesUtils.writeIndexName(model, metric.getTimeBucket());
-                                 // Format the name to follow the global physical index naming policy.
-                                 if (!IndicesMetadataCache.INSTANCE.isExisting(
-                                     getClient().formatIndexName(indexName))) {
-                                     indexName = IndexController.INSTANCE.getTableName(model);
-                                 }
-                                 return indexName;
+                                 groupIndices.computeIfAbsent(indexName, v -> new ArrayList<>()).add(metric);
                              } else {
                                  // Metadata level metrics, always use alias name, due to the physical index of the records
                                  // can't be located through timestamp.
-                                 return IndexController.INSTANCE.getTableName(model);
+                                 String indexName = IndexController.INSTANCE.getTableName(model);
+                                 groupAliasIndices.computeIfAbsent(indexName, v -> new ArrayList<>()).add(metric);
                              }
-                         })
-                     );
-
+                         });
         // The groupIndices mostly include one or two group,
         // the current day and the T-1 day(if at the edge between days)
```

Comment:
I think this comment is not correct anymore. The alias is only used for `!isTimeRelativeID` entities only, right? It doesn't matter at the day edge or not anymore.